### PR TITLE
Updated to use .NET Core 2.1 (etc.)

### PR DESCRIPTION
**Pull request by petrsvihlik**
### Motivation

- Updated to use .net core 2.1
- Updated to use latest nugets, especially kenticocloud.delivery. 
- Refactored app's entry point to fully utilize DI.
- Refactored configuration loading.
- Adjusted build procedure in order to prepare for #66 

### Checklist

- [x] Code follows coding conventions held in this repo
- [x] Automated tests have been added
- [x] Tests are passing
- [x] Docu has been updated (if applicable)
- [x] Temporary settings (e.g. project ID used during development and testing) have been reverted to defaults

### How to test

If manual testing is required, what are the steps?

### CODE REVIEW

File: src/CloudModelGenerator/CodeGenerator.cs [hunk 1]
```diff
@@ -25,9 +32,22 @@ public CodeGenerator(IOptions<CodeGeneratorOptions> options)
...
+            }
+
+            if (!string.IsNullOrEmpty(_options.BaseClass))
+            {
+                GenerateBaseClass();
+            }
+            return 0;
         }
 
         public void GenerateContentTypeModels(bool structuredModel = false)
```
Comment (makma) [inline src/CloudModelGenerator/CodeGenerator.cs]: Does public make sense when it's encapsulated in Run() ?

File: src/CloudModelGenerator/Program.cs [hunk 2]
```diff
@@ -1,128 +1,93 @@
...
-            Configuration = builder.Build();
-
-            CodeGeneratorOptions options = new CodeGeneratorOptions();
-
-            // Load Code Generator Options from the configuration sources
-            new ConfigureFromConfigurationOptions<CodeGeneratorOptions>(Configuration).Configure(options);
-
-            // Load Delivery Options from the configuration sources
-            new ConfigureFromConfigurationOptions<DeliveryOptions>(Configuration.GetSection(nameof(DeliveryOptions))).Configure(options.DeliveryOptions);
+            var unexpectedArgs = new List<string>(syntax.RemainingArguments);
```
Comment (makma) [inline src/CloudModelGenerator/Program.cs]: I really like all this args syntax handling! 😻

File: test/CloudModelGenerator.Tests/CodeGeneratorTests.cs [hunk 3]
```diff
@@ -21,6 +23,37 @@ public CodeGeneratorTests()
...
+            var mockOptions = new Mock<IOptions<CodeGeneratorOptions>>();
+            var options = new CodeGeneratorOptions
+            {
+                OutputDir = ""
+            };
+            mockOptions.Setup(x => x.Value).Returns(options);
+            var mockClient = new Mock<IDeliveryClient>();
+
+            var codeGenerator = new CodeGenerator(mockOptions.Object, mockClient.Object);
+            Assert.Equal(expectedOutputDir, expectedOutputDir);
```
[the remark this answers is not in the document]
Comment (petrsvihlik) [reply test/CloudModelGenerator.Tests/CodeGeneratorTests.cs]: yes :)

File: test/CloudModelGenerator.Tests/CodeGeneratorTests.cs [hunk 4]
```diff
@@ -34,14 +67,14 @@ public void IntegrationTest(bool cmApi)
...
             {
-                DeliveryOptions = { ProjectId = "975bf280-fd91-488c-994c-2f04416e5ee3" },
                 Namespace = "CustomNamespace",
                 OutputDir = TEMP_DIR,
                 ContentManagementApi = cmApi
             });
 
-            var codeGenerator = new CodeGenerator(mockOptions.Object);
-            codeGenerator.Client.HttpClient = httpClient;
+            var client = DeliveryClientBuilder.WithProjectId("975bf280-fd91-488c-994c-2f04416e5ee3").WithHttpClient(httpClient).Build();
```
[the remark this answers is not in the document]
Comment (petrsvihlik) [reply test/CloudModelGenerator.Tests/CodeGeneratorTests.cs]: considered :)

**Pull request merged**